### PR TITLE
Added config option to change tab title alignment

### PR DESCRIPTION
**Pull request by dylanaraps**
This pull request adds a config option to change the tab title alignment. 

Once again, thanks to The Compiler for all the help!

### CODE REVIEW

File: qutebrowser/config/configdata.py [hunk 1]
```diff
@@ -625,6 +625,10 @@ def data(readonly=False):
              "* `{id}`: The internal tab ID of this tab.\n"
              "* `{scroll_pos}`: The page scroll position."),
 
+            ('title-alignment',
+             SettingValue(typ.TitleAlignment(), 'left'),
+             "Alignment of the text inside of tabs)"),
```
Comment (The-Compiler) [inline qutebrowser/config/configdata.py]: that `)` doesn't belong there :wink:

Comment (dylanaraps) [inline qutebrowser/config/configdata.py]: oops

File: qutebrowser/config/configtypes.py [hunk 2]
```diff
@@ -1336,6 +1336,21 @@ def __init__(self, none_ok=False):
             valid_values=ValidValues('top', 'bottom', 'left', 'right'))
 
 
+class TitleAlignment(MappingType):
+
+    """Alignment of tab text"""
```
Comment (The-Compiler) [inline qutebrowser/config/configtypes.py]: I'd call this `TextAlignment` and change the docstring accordingly - I think this makes sense as a generic type and could be used for other config options in the future too.

**Comment by dylanaraps** [comment]
I've updated the pull request.

**Comment by The-Compiler** [comment]
Looks good now - I'll test and merge this tomorrow, need some sleep now :wink:

Thanks!

**Comment by dylanaraps** [comment]
No problem!

**Comment by The-Compiler** [comment]
This seems to have caused some test issues:

```
 TestTabWidget.test_small_icon_doesnt_crash 
CALL ERROR: Qt exceptions in virtual methods:

  File "/home/travis/build/The-Compiler/qutebrowser/qutebrowser/mainwindow/tabwidget.py", line 570, in drawControl
    self.drawControl(QStyle.CE_TabBarTabLabel, opt, p, widget)
  File "/home/travis/build/The-Compiler/qutebrowser/qutebrowser/mainwindow/tabwidget.py", line 581, in drawControl
    Qt.AlignVCenter | Qt.TextHideMnemonic)
  File "/home/travis/build/The-Compiler/qutebrowser/qutebrowser/config/config.py", line 128, in get
    return objreg.get('config').get(*args, **kwargs)
  File "/home/travis/build/The-Compiler/qutebrowser/tests/helpers/stubs.py", line 367, in get
    raise configexc.NoOptionError(opt, sect)
NoOptionError: No option 'title-alignment' in section 'tabs'

```
 test_progress_affecting_statusbar_height 
tests/unit/mainwindow/statusbar/test_progress.py:120: Failure: Qt messages with level WARNING or above emitted
--------- Captured Qt messages ---------
painting/qpaintdevice.cpp:virtual QPaintDevice::~QPaintDevice():55:
    QtWarningMsg: QPaintDevice: Cannot destroy paint device that is being painted
```

```
tests/unit/misc/test_readline.py::test_rl_forward_word[<one> two-one two|] Fatal Python error: Segmentation fault
Current thread 0x00007ff237c2f740 (most recent call first):
  File "/home/travis/build/The-Compiler/qutebrowser/.tox/unittests-nodisp/lib/python3.4/site-packages/py/_code/source.py", line 348 in getstatementrange_ast
  File "/home/travis/build/The-Compiler/qutebrowser/.tox/unittests-nodisp/lib/python3.4/site-packages/py/_code/code.py", line 204 in getsource
  File "/home/travis/build/The-Compiler/qutebrowser/.tox/unittests-nodisp/lib/python3.4/site-packages/py/_code/code.py", line 455 in _getentrysource
  File "/home/travis/build/The-Compiler/qutebrowser/.tox/unittests-nodisp/lib/python3.4/site-packages/py/_code/code.py", line 528 in repr_traceback_entry
  File "/home/travis/build/The-Compiler/qutebrowser/.tox/unittests-nodisp/lib/python3.4/site-packages/py/_code/code.py", line 582 in repr_traceback
  File "/home/travis/build/The-Compiler/qutebrowser/.tox/unittests-nodisp/lib/python3.4/site-packages/py/_code/code.py", line 590 in repr_excinfo
  File "/home/travis/build/The-Compiler/qutebrowser/.tox/unittests-nodisp/lib/python3.4/site-packages/py/_code/code.py", line 412 in getrepr
  File "/home/travis/build/The-Compiler/qutebrowser/.tox/unittests-nodisp/lib/python3.4/site-packages/_pytest/main.py", line 404 in _repr_failure_py
  File "/home/travis/build/The-Compiler/qutebrowser/.tox/unittests-nodisp/lib/python3.4/site-packages/_pytest/python.py", line 744 in _repr_failure_py
  File "/home/travis/build/The-Compiler/qutebrowser/.tox/unittests-nodisp/lib/python3.4/site-packages/_pytest/python.py", line 751 in repr_failure
  File "/home/travis/build/The-Compiler/qutebrowser/.tox/unittests-nodisp/lib/python3.4/site-packages/_pytest/runner.py", line 224 in pytest_runtest_makereport
  File "/home/travis/build/The-Compiler/qutebrowser/.tox/unittests-nodisp/lib/python3.4/site-packages/_pytest/vendored_packages/pluggy.py", line 596 in execute
[...]
```

I don't understand the second and third one. For the first one, it seems the tests in `tests/unit/mainwindow/test_tabwidget.py` need to have this config option added.

If you look at that file, there's a fake config in `CONFIG`. Can you add `'title-alignment': 'left',` or so under `tabs` there?

The style linters also showed some nitpicks, I'll do line comments for them.

If you want to run the test suite locally, install `tox` for python (e.g. `python-tox` on Debian/Ubuntu), and run `tox` inside the qutebrowser repo. Otherwise, you can just push and Travis CI will re-run the checks.

File: qutebrowser/config/configtypes.py [hunk 3]
```diff
@@ -1336,6 +1336,21 @@ def __init__(self, none_ok=False):
             valid_values=ValidValues('top', 'bottom', 'left', 'right'))
 
 
+class TextAlignment(MappingType):
+
+    """Alignment of text"""
```
Comment (The-Compiler) [inline qutebrowser/config/configtypes.py]: Please add a period after `text` to satisfy the pep257 checker :wink:

File: qutebrowser/config/configtypes.py [hunk 4]
```diff
@@ -1336,6 +1336,21 @@ def __init__(self, none_ok=False):
...
+        'left': Qt.AlignLeft,
+        'right': Qt.AlignRight,
+        'center': Qt.AlignCenter,
+    }
+
+    def __init__(self, none_ok=False):
+        super().__init__(
+            none_ok,
+            valid_values=ValidValues('left', 'right', 'center'))
+
```
Comment (The-Compiler) [inline qutebrowser/config/configtypes.py]: Please add another blank line here (2 blank lines between classes, 1 between methods)

**Comment by dylanaraps** [comment]
Thanks for the comments.

 I've fixed the issues above and I'm just running tox now.

File: tests/unit/mainwindow/test_tabwidget.py [hunk 5]
```diff
@@ -44,6 +44,7 @@ class TestTabWidget:
             'indicator-width': 3,
             'indicator-padding': configtypes.PaddingValues(2, 2, 0, 4),
             'title-format': '{index}: {title}',
+            'title-alignment': 'left',
```
Comment (The-Compiler) [inline tests/unit/mainwindow/test_tabwidget.py]: Whoops, my bad - this needs to be the already parsed value, i.e. `Qt.AlignLeft` (you'll also need to add `from PyQt5.QtCore import Qt` to the imports at the top).

**Comment by dylanaraps** [comment]
Done!

**Comment by The-Compiler** [comment]
It should be `Qt.AlignLeft` as a value, not `'Qt.AlignLeft'` as a string :wink:

**Comment by dylanaraps** [comment]
Doh!

Fixed, thanks for all the help!

**Comment by The-Compiler** [comment]
Looks good now! I plan to test and merge this later today.

**Comment by dylanaraps** [comment]
Awesome! 

Thanks again.

**Pull request merged**